src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/Args.java

```diff
@@ -126,13 +125,27 @@
    * @return [key1=value1,key2=value2,...], [] if keyValueMapping=empty/null
    */
   public static List<String> keyValues(Map<?, ?> keyValueMapping) {
+    return namedKeyValues("", keyValueMapping);
+  }
+
+  /**
+   * Produces a named key/value pair list given a name and a {@link Map}.
+   *
+   * @return [--name, key1=value1, --name, key2=value2,...], [] if keyValueMapping=empty/null
+   */
+  public static List<String> namedKeyValues(String name, Map<?, ?> keyValueMapping) {
     List<String> result = Lists.newArrayList();
```

Comment:
I'm really not a fan of having namedKeyValues method called from the non-named use-case and branching on null/empty name. 

I've just seen that zip is not in jdk8 (learn something new every day).  This looks like a tidy use-case for streams and mapping, but the weirdness of keyValues stymies elegance. what do you tihnk of just ignoring keyValues to its confusion and implement flaggedKeyValues with something as simple as: 
```
keyValues().entrySet().stream()
        .map(e -> "--flag " + e.getKey() + '=' + e.getValue())
        .collect(Collectors.joining(" , "));
```

is that crazy?